src/mixins/createCachedRenderMixin.ts

```diff
@@ -121,9 +126,15 @@ const createCachedRenderMixin: CachedRenderFactory = createStateful
 					props[key] = cachedRender.listeners[key];
 				}
 				const classes: { [index: string]: boolean; } = {};
+				const widgetClasses: string[] = widgetClassesMap.get(cachedRender) || [];
```

Comment:
Instead of a `WeakMap` let each widget instance manage and own their array of previous classes?